norduser/group_monitor.go

```diff
@@ -211,19 +286,22 @@ func (n *NordvpnGroupMonitor) Start() error {
 			// Because utilities used to modify the group do so atomically, we also need to monitor for creation of
 			// the file instead of modifications.
 			if (event.Has(fsnotify.Create) || event.Has(fsnotify.Write)) && event.Name == groupfilePath {
-				newGroupMembers, err := getNordvpnGroupMembers()
+				newGroupMembers, err := getNordVPNGroupMembers()
+				newGroupMembers = updateGroupMembersState(newGroupMembers, getSessions())
 				if err == nil {
-					n.handleGroupUpdate(currentGrupMembers, newGroupMembers)
-					currentGrupMembers = newGroupMembers
+					currentGrupMembers = n.handleGroupUpdate(currentGrupMembers, newGroupMembers)
 				} else {
-					log.Println("Failed to get new group members:", err)
+					log.Println(internal.ErrorPrefix, "Failed to get new group members:", err)
 				}
 			}
 		case err, ok := <-watcher.Errors:
 			if !ok {
 				return fmt.Errorf("groupfile monitor error channel closed")
 			}
 			log.Println(internal.ErrorPrefix, "group monitor error:", err)
+		case <-time.After(10 * time.Second):
+			newGroupMembers := updateGroupMembersState(currentGrupMembers, getSessions())
```

Comment:
Executing an external process every 10 seconds all the time when `nordvpnd` is running is something we certainly should not do.

Reply:
Maybe using fsnotify to monitor the `/var/run/utmp` would be some kind of solution to trigger on session change events without constantly polling in a loop?

Reply:
Here is a `inotify` sample: https://github.com/keliramu/watch-socket/blob/main/server/main.go#L82
without additional dependencies.
Maybe would help.